src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,142 @@
+        private static async Task<ISiloHost> StartSilo()
+        {
+            // define the cluster configuration
+            var builder = new SiloHostBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .Configure<EndpointOptions>(options => options.AdvertisedIPAddress = IPAddress.Loopback)
+                .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(HelloGrain).Assembly).WithReferences())
+                .ConfigureLogging(logging => logging.AddConsole());
+
+            var host = builder.Build();
+            await host.StartAsync();
+            return host;
+        }
+```
+
+| Option | Used for |
+|-------------|----------|
+| `.UseLocalhostClustering()` | Declaring that we are using a single local silo |
+| `ClusterOptions` | ClusterId is the name for the Orleans cluster must be the same for silo and client so they can talk to each other. ServiceId is the ID used for the application and it must not change across deployments|
+| `EndpointOptions` | This tells the silo where to listen. For this example, we are using a `loopback`. |
+| `ConfigureApplicationParts` | Adds the assembly with grain classes to the application setup. |
+
+After loading the configurations, the SiloHost is built and then started asynchronously.
+
+## Configuring the client
+
+Similar to the silo, the client is configured via `ClientBuilder` and a similar collection of option classes.
+
+```csharp
+        private static async Task<IClusterClient> StartClientWithRetries()
+        {
+            attempt = 0;
+            IClusterClient client;
+            client = new ClientBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .ConfigureLogging(logging => logging.AddConsole())
+                .Build();
+
+            await client.Connect(RetryFilter);
+            Console.WriteLine("Client successfully connect to silo host");
+            return client;
+        }
+
+```
+
+| Option | Used for |
+|-------------|----------|
+| `.UseLocalhostClustering()` | Same as for SiloHost |
+| `ClusterOptions` | Same as for SiloHost |
+
+A more in-depth guide to configuring your client can be found [in the Client Configuration section of the Configuration Guide.](http://dotnet.github.io/orleans/Documentation/clusters_and_clients/configuration_guide/client_configuration.html)
+
+## Writing a grain
+
+Grains are the building blocks of an Orleans application, and you can read more about them in the [Core Concepts section of the Orleans documentation.](http://dotnet.github.io/orleans/Documentation/core_concepts/index.html)
```

Comment:
Can we put the whole first paragraph here : 

Grains are the key primitives of the Orleans programming model. Grains are the building blocks of an Orleans application, they are atomic units of isolation, distribution, and persistence. Grains are objects that represent application entities. Just like in the classic Object Oriented Programming, a grain encapsulates state of an entity and encodes its behavior in the code logic. Grains can hold references to each other and interact by invoking each other’s methods exposed via interfaces. 

IMO We need to at least give an overview here to get people's eyes on what grains are and catch their attention. Otehrwise this single line doesn't really give much attraction or useful info.

Reply:
Yes I have added the paragraph (and kept the link.)